modules/mask.py

```diff
@@ -1,153 +1,95 @@
 import numpy as np
-import numpy.ma as ma
-import xarray as xr
-
-def load_land_mask(data_path):
-    """Loads land fraction file from
-    http://climate-modelling.canada.ca/climatemodeldata/canrcm/CanRCM4/NAM-44_ECMWF-ERAINT_evaluation/fx/atmos/sftlf/index.shtml
-    and creates a mask for cells that are 0% land.
-    ----------------------------------------------
-    Args:
-        data_path (str): path to the land fraction mask file
-    Returns:
-        mask (numpy.masked_array): masked array based on file
-    """
-    ds_mask = xr.open_dataset(data_path)
-    mask = ma.masked_greater(ds_mask['sftlf'].values, 10.).mask
-    return mask
-
-def mask_land_and_nan(dv_field, mask_land):
-    """Creates a master mask by masking any cell in the ensemble
-    that has a NaN value. If a NaN value is found in the ensemble,
-    that grid cell is masked for every ensemble member, thus
-    disqualifying that grid cell from further analysis. It is then
-    combined with the land mask, to create a master mask of cells
-    to use in the analysis.
-    ------------------------------------------------------------
-    Args:
-        dv_field (numpy.ndarray): datacube containing the ensemble members
-        mask_land (numpy.masked_array): land mask created by load_land_mask()
-
-    Returns:
-        mask_master (numpy.ndarray): a CanRCM4 field shaped array containing
-            a boolean master mask
-    """
-    nan_mask = np.apply_over_axes(
-                    np.logical_or.reduce,
-                    np.isnan(dv_field),
-                    (0)
-    )
-
-    mask_master = ~np.logical_or(~mask_land, nan_mask)
-
-    return mask_master
-
-def flatten_mask(mask_master):
-    """Reshapes the master mask created in mask_master into
-    the flattened ensemble shape, that is, (number of grid cells) x (number of
-    ensemble members). Each row in this new shape represents a grid cell
-    from the ensemble members.
-
+from shapely.geometry import Point
+import geopandas as gpd
+
+
+def check_polygon(p):
```

Comment:
A suggestion on naming: all of these functions named `check_*` have names that are not very precise. I.e. from the function name alone, it's not clear what the function does. So it *checks* some property, but what is it checking for? I had to read the function to find out (which is problematic for reading the test code... it took digging to find out what the tests were actually testing for).

Try to expand the the function name to succinctly encapsulate what it does. And at the very least, but a longer description in a [docstring](https://docs.python-guide.org/writing/documentation/#docstrings-versus-block-comments) not a comment.